src/console_results.sh

```diff
@@ -0,0 +1,23 @@
+export renderResult
+
+function renderResult() {
+  local totalTests=$1
+  local totalPassed=$2
+  local totalFailed=$3
+
+  echo ""
+  local totalAssertions=$((totalPassed + totalFailed))
+  printf "\
+${COLOR_FAINT}Total tests:${COLOR_DEFAULT} ${COLOR_BOLD}${totalTests}${COLOR_DEFAULT}
+${COLOR_FAINT}Total assertions:${COLOR_DEFAULT} ${COLOR_BOLD}${totalAssertions}${COLOR_DEFAULT}\n"
+
+  if [ "$totalFailed" -gt 0 ]; then
+    printf "${COLOR_FAINT}Total assertions failed:${COLOR_DEFAULT} ${COLOR_BOLD}${COLOR_FAILED}${totalFailed}${COLOR_DEFAULT}\n"
+    exit 1
+  else
+    printf "${COLOR_ALL_PASSED}All assertions passed.${COLOR_DEFAULT}\n"
+  fi
+}
+
+# Set a trap to call render_result when the script exits
```

Comment:
```suggestion
# Set a trap to call renderResult when the script exits
```

Reply:
Related to this, the convention on Bash it's to use for the variables always Upper snake case, so my suggestion it's to add a https://github.com/Chemaclass/bashunit/issues/20 static analysis tool to the local repo and prepare a pre-commit for it.

A conversation about this is: https://unix.stackexchange.com/questions/42847/are-there-naming-conventions-for-variables-in-shell-scripts#answer-42849

Although for example, this is not in these guidelines https://google.github.io/styleguide/ the community uses this